parsec/core/cli/bootstrap_organization.py

```diff
@@ -1,75 +1,46 @@
-        save_device_with_password(config_dir, device, password, force=force)
-
-    now = pendulum.now()
-    user_certificate = UserCertificateContent(
-        author=None,
-        timestamp=now,
-        user_id=device.user_id,
-        public_key=device.public_key,
-        profile=device.profile,
-    ).dump_and_sign(root_signing_key)
-    device_certificate = DeviceCertificateContent(
-        author=None, timestamp=now, device_id=device_id, verify_key=device.verify_key
-    ).dump_and_sign(root_signing_key)
-
-    async with spinner(f"Sending {device_display} to server"):
-        async with apiv1_backend_anonymous_cmds_factory(organization_bootstrap_addr) as cmds:
-            await cmds.organization_bootstrap(
-                organization_bootstrap_addr.organization_id,
-                organization_bootstrap_addr.token,
-                root_verify_key,
-                user_certificate,
-                device_certificate,
+    async with apiv1_backend_anonymous_cmds_factory(addr=addr) as cmds:
+        async with spinner("Bootstrapping organization in the backend"):
+            new_device = await do_bootstrap_organization(
+                cmds=cmds, human_handle=human_handle, device_label=device_label
             )
 
-    organization_addr_display = click.style(organization_addr.to_url(), fg="yellow")
-    click.echo(f"Organization url: {organization_addr_display}")
+        device_display = click.style(new_device.slughash, fg="yellow")
+        with operation(f"Saving device {device_display}"):
+            save_device_with_password(
+                config_dir=config.config_dir, device=new_device, password=password, force=force
+            )
 
 
 @click.command(short_help="configure new organization")
-@click.argument("device", type=DeviceID, required=True)
-@click.option("--addr", "-B", type=BackendOrganizationBootstrapAddr.from_url, required=True)
-@click.option("--config-dir", type=click.Path(exists=True, file_okay=False))
+@core_config_options
+@click.argument("addr", type=BackendOrganizationBootstrapAddr.from_url)
+@click.password_option(prompt="Choose a password for the device")
 @click.option("--force", is_flag=True)
-@click.password_option()
-def bootstrap_organization(device, addr, config_dir, force, password):
+def bootstrap_organization(config, addr, password, force, **kwargs):
     """
     Configure the organization and register it first user&device.
     """
-
-    config_dir = Path(config_dir) if config_dir else get_default_config_dir(os.environ)
-    debug = "DEBUG" in os.environ
-    configure_logging(log_level="DEBUG" if debug else "WARNING")
-
-    with cli_exception_handler(debug):
-        trio_run(_bootstrap_organization, debug, device, addr, config_dir, force, password)
+    with cli_exception_handler(config.debug):
+        # Disable task monitoring given user prompt will block the coroutine
+        trio_run(_bootstrap_organization, config, addr, password, force)
```

Comment:
I'm a bit confused by those two `bootstrap_organizations` commands. It seems like there is one legacy (APIV1) command and one new command, but they both use the `apiv1_backend_anonymous_cmds_factory`, is that normal?

Also, how does one decide when to use which? Does this have to do with the backend version? In this case, why do we want to support old backend versions?

Reply:
see https://github.com/Scille/parsec-cloud/pull/1174#discussion_r440265323

tl;dr: apiv1 is only used for administration authentications (typically organization create/stats) and for organization bootstrap